core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -1006,12 +1006,28 @@ public static void udpateUserRoleMappingInBatchMode(Connection dbConnection, Str
                         + Arrays.toString(count));
             }
             dbConnection.commit();
-        } catch (SQLException e) {
-            String errorMessage = "Using sql : " + sqlStmt + " " + e.getMessage();
-            if (log.isDebugEnabled()) {
-                log.debug(errorMessage, e);
+
+        } catch (SQLException e ) {
+            boolean isUniqueKeyViolationException = false;
+            for (Throwable cause : e) {
```

Comment:
Shall we recheck this line, whether it's possible to iterate through the exceptions?

Reply:
Since the SQLException class implements `Iterable<Throwable>`, exception can be iterable.